ion-schema/src/internal_traits.rs

```diff
@@ -59,16 +60,51 @@ pub(crate) struct SchemaStore {
 ///
 /// Implementations of `WriteAsIon` may delegate to this when possible.
 pub(crate) trait WriteAsIsl<V: IslVersion> {
-    fn write_as_isl<W: ValueWriter>(&self, writer: W) -> IonSchemaResult<()>;
+    fn write_as_isl<W: ValueWriter>(
+        &self,
+        writer: W,
+        ctx: &WriteContext<V>,
+    ) -> IonSchemaResult<()> {
+        let (major, minor) = V::MAJOR_MINOR;
+        invalid_schema_error(format!(
+            "Cannot write to Ion Schema Language {major}.{minor}"
```

Comment:
```suggestion
            "Cannot write {self:?} as Ion Schema Language {major}.{minor}"
```
Or maybe even more user-friendly: `{self:?} is not supported in Ion Schema Language {major}.{minor}`.